src/Module/Diaspora/Fetch.php

```diff
@@ -44,10 +44,7 @@ public static function rawContent()
 
 				if (Strings::normaliseLink($host) != Strings::normaliseLink($app->getBaseURL())) {
 					$location = $host . "/fetch/" . $app->argv[1] . "/" . urlencode($guid);
-
-					header("HTTP/1.1 301 Moved Permanently");
-					header("Location:" . $location);
-					exit();
+					$app->redirect($location);
```

Comment:
Right, but we lost the `301` status code in the process, maybe this should be a additional parameter of `System::externalRedirect`?.